[clarud] iP

clarud opened the pull request:
# Weeny
> Hi how can I help you? - _Weeny_

Are you getting overwhelmed by the ALL of the tasks on your hands? 
~~Yes 😭~~
Fret not Weeny task manager is here to help! 😇

What does it do?

- Manage your multitude of tasks
- Organise your task
- Set deadlines

Current commands:

1. `todo` : Add todo task
2. `event` : Add event task
3. `deadline` : Add deadline task
4. `delete` : Delete tasks from your list
5. `mark` : Mark tasks as done
6. `unmark` : Mark tasks as undone
7. `list` : List all tasks
8. `find` : Look for tasks that match the keyword

Some basic command examples are:

```
todo Buy groceries
deadline Math assignment /by 12/09/2024 2359
event Movie with mom! /from 03/09/2024 2000 /to 03/09/2024 2200
```

Download

Click this [**link**](https://nus-cs2103-ay2425s1.github.io/website/) to download!

Features:

- [x] Managing task
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
> [!CAUTION]
> More features coming soon

Here's the `main` method:
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

clarud committed:
Level-0

clarud committed:
Level-1

clarud committed:
Level-2

clarud committed:
Level-3

clarud committed:
Level-4

clarud committed:
A-TextUiTesting

clarud committed:
Level-5

clarud committed:
Level-6

clarud committed:
A-Enums

clarud committed:
Add data directory and task file reading and creating feature

clarud committed:
Add feature to format and write data to save task list

clarud committed:
Merge branch 'branch-Level-7'
Add feature to save task list data

clarud committed:
Add feature to handle time conversion

clarud committed:
Merge Level-8 into main

clarud committed:
Add parser class

clarud committed:
Add Ui class

clarud committed:
Add storage class

clarud committed:
Clean up code

clarud committed:
Add tasklist class

clarud committed:
Merge branch-A-MoreOOP with master

clarud committed:
Put all files into a package

clarud committed:
Merge origin/add-gradle-support to master

clarud committed:
Add gradle features'

clarud committed:
Add test cases

clarud committed:
Create a JAR file

clarud committed:
Add Javadoc to all files

clarud committed:
Fix coding standards

clarud committed:
Add search keyword feature

clarud committed:
Fix coding standard

clarud committed:
Merge branch-A-CodingStandard into master

Merge branch-A-CodingStandard into master

clarud committed:
Merge branch-Level-9 into master

clarud committed:
Add Checkstyle feature

clarud committed:
Add Checkstyle feature

clarud committed:
Add checkstyle features

clarud committed:
Merge branch A-Checkstyle into master

clarud committed:
Modify packages

clarud committed:
Modify packages

clarud committed:
Add simple GUI

clarud committed:
Add simple GUI

noth-k commented on src/main/java/Weeny/Ui.java:
```diff
@@ -0,0 +1,134 @@
+     * @param task The task that was deleted.
+     * @param size The updated number of tasks.
+     */
+    public void showTaskDeletedMessage(Task task, int size) {
```
there seems to be a inconsistency in naming, the previous function was "print" but this one appears to be "show". Maybe they could be similar for consistency?

noth-k commented on src/main/java/Weeny/Task.java:
```diff
@@ -0,0 +1,89 @@
+     * @param keyWord
+     * @return True if the keyword is found in the description
+     */
+    public boolean containString(String keyWord) {
```
could this be "containsString" perhaps?

noth-k commented on src/main/java/Weeny/Task.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return A string with task details.
+     */
+    public String toOutput() {
```
Could the naming for this function be more clear?

noth-k reviewed (commented):
Overall good naming conventions, using questions for booleans, verbs for methods etc. Nice job :)

clarud committed:
Seperate WeenyGui out from Weeny

clarud committed:
Seperate WeenyGui out from Weeny

clarud committed:
Seperate WeenyGui out from Weeny

clarud committed:
Add fxml

clarud committed:
Add fxml

clarud committed:
Delete unnecessary file

limjeremy496 commented on src/main/java/Weeny/task/Events.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Represents an event task with a description, start, and end date/time.
+ */
+public class Events extends Task {
```
Should the class name be singular rather than plural since one Event is created?

limjeremy496 commented on src/main/java/Weeny/task/Deadlines.java:
```diff
@@ -0,0 +1,52 @@
+/**
+ * Represents a deadline task with a description and due date/time.
+ */
+public class Deadlines extends Task {
```
Should the class name be singular rather than plural since one Deadline is created?

limjeremy496 commented on src/main/java/Weeny/Weeny.java:
```diff
@@ -0,0 +1,181 @@
+     * @throws IllegalArgumentException if the input is too short.
+     */
+    private static void validateTodoInput(String input) {
+        if (input.length() <= 5) {
```
What do you think about defining constants for the different length constraints?

limjeremy496 commented on src/main/java/Weeny/parser/Parser.java:
```diff
@@ -0,0 +1,100 @@
+     * @param date Date string to convert.
+     * @return The corresponding LocalDate object.
+     */
+    public LocalDate convertDate(String date) {
```
Will it be ideal to use a try-catch for incorrect date formats?

limjeremy496 reviewed (commented):
Overall very nicely done :)

clarud committed:
Merge branch-Level-10 into master

clarud committed:
Add assertions at key locations

Lack of areas to check when the code gets long.
It is common to miss out on bugs which can cause the program to fail.

Write assertions to important areas allows us to ensure our
assumptions is correct. For example, the tasklist file is created by the
code.

Using assertions ensure that the intended effect of the methods is done
correctly.

clarud committed:
Merge pull request #1 from clarud/branch-A-Assertions

Add assertions at key locations

clarud committed:
Improve on code quality

Readability on some parts of the code can be better.
Presence of unused code.

Add constants to date and time format.
Add Javadoc for GUI related methods and classes.

Improve readability and flow of code.

clarud committed:
Merge pull request #2 from clarud/branch-A-CodeQuality

Improve on code quality

clarud committed:
Add streams to some methods

Current code is long and can be more efficient.

Add streams to find method.

Streamline searching keyword process.

clarud committed:
Merge pull request #3 from clarud/branch-A-Streams

Add streams to some methods

clarud committed:
Add schedule command

Gives a list of tasks that happens on a date.
Todo is always given, Deadline if before end date and Event if within
event date.
Tasks are sorted by time to give a timeline on a date

clarud committed:
Merge branch 'branch-Schedule'

clarud committed:
Change images for user and weeny

clarud committed:
Add icon for send button

clarud committed:
Improve styling and functionality of GUI

Add styling to GUI
Incorporate role swap distinguishing feature

clarud committed:
Merge branch 'branch-BetterGUI'

clarud committed:
Modify display messages to be more personal

clarud committed:
Merge branch 'branch-A-Personality'

clarud committed:
Add checks for new commands

clarud committed:
Add error exception for invalid event time

clarud committed:
Add more error handling

clarud committed:
Rectify logic issues in schedule

clarud committed:
Fix validation issue for some commands

clarud committed:
Add product information

clarud committed:
Remove comments from README

clarud committed:
Add new JUnit test

clarud committed:
Fix fxml compatibility warning

clarud committed:
Improve styling for README

clarud committed:
Add screenshot to README

clarud committed:
Add files via upload

clarud committed:
Delete Weeny.jar
